frontend/src/components/modules/GameInterface/GameInterface.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useState, useEffect } from "react";
+import { Typography, Box } from "@mui/material";
+import PointTable from "./PointTable";
+import Timer from "./Timer";
+import OfficialButtons from "./OfficialButtons";
+import TimerButton from "./TimerButton";
+import api from "api/axios";
+import { useParams } from "react-router-dom";
+import { type AddPointRequest } from "types/requests";
+import type { PointType, PlayerColor, Match, MatchPlayer } from "types/models";
+import "./GameInterface.css";
+import { useAuth } from "context/AuthContext";
+import { joinMatch, leaveMatch } from "sockets/emit";
+import { useSocket } from "context/SocketContext";
+
+export interface matchData {
+  timerTime: number;
+  players: MatchPlayer[];
+  playerNames: string[];
+  winner: string | undefined;
+  officials: string;
+}
+
+const GameInterface: React.FC = () => {
+  const [matchInfo, setMatchInfo] = useState<matchData>({
+    timerTime: 300,
+    players: [],
+    playerNames: [],
+    winner: undefined,
+    officials: ""
+  });
+
+  useEffect(() => {
+    if (matchId !== undefined) {
```

Comment:
pass the matchId as dependency to this useEffect

Reply:
Probably also needs a check if they have joined the room if it's passed to the dependency array since it will attempt to join the room on every render then.